enhancements/kubernetes-nmstate-integration.md

```diff
@@ -0,0 +1,165 @@
+---
+title: kubernetes-nmstate
+authors:
+  - "@schseba"
+reviewers:
+
+approvers:
+  - TBD
+  
+creation-date: 2019-12-18
+last-updated: 2019-12-18
+status: 
+---
+
+# kubernetes-nmstate
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+A proposal to deploy [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate/) on OpenShift.
+
+Node-networking configuration driven by Kubernetes and executed by
+[nmstate](https://nmstate.github.io/).
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+The [Container Network Interface](https://github.com/containernetworking/cni)
+(CNI) standard enables different
+solutions for connecting networks on the node with pods. Some of them are
+[part of the standard](https://github.com/containernetworking/plugins), and there are
+others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni),
+[SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the
+pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster,
+with dynamic networking requirements, is a challenge by itself - and this is
+what this project is addressing.
+
+### Goals
+
+- Deploy kubernetes-nmstate as part of openshift
+
+### Non-Goals
+
+- Replace SRIOV operator
+
+## Proposal
+
+A new kubernetes-nmstate operator Deployment is deployed to the cluster as part of the the OpenShift
+installation, via CVO.
+
+The operator has a single CRD, called `NMstate`. When a custom resource of kind `NMstate`, with 
+a name of 'nmstate' is created, it will then create the CRDs for the kubernetes-nmstate handler.
+Then, the namespace, RBAC, and finally the DaemonSet are applied. Custom resources of kind `NMstate`
+with other names will be ignored.
+
+A new kubernetes-nmstate handler DaemonSet is deployed in the cluster by the operator.
```

Comment:
Maybe it worth mentioning that Operator CR spec supports filtering on which nodes the DaemonSet will be deployed.

Reply:
please explain how this component relates to https://github.com/openshift/enhancements/pull/344